systest/Makefile

```diff
@@ -1,119 +1,178 @@
+
+# /ABS/PATH/PROJECT/systest/scripts
+SCRIPT_DIR := $(MAKEFILE_DIR)/scripts
+
+# /ABS/PATH/PROJECT
+PROJDIR := $(shell dirname $(realpath $(MAKEFILE_DIR)))
+
+# PROJECT
+PROJNAME := $(shell basename $(PROJDIR))
 
 # - <nearest reachable tag>-<num commits since>-g<abbreviated commit id>
-version := $(shell git describe --long)
-timestamp ?= $(shell date +"%Y%m%d-%H%M%S")
-export timestamp   # Only eval timestamp in the top make.
-branch := $(shell git rev-parse --abbrev-ref HEAD)
-
-RESULTSDIR := $(MAKEFILE_DIR)/test_results/$(PROJECT)/$(PROJECT)_$(branch)
-
-unit_session := unit_$(version)_$(timestamp)
-unit_results := $(RESULTSDIR)/$(unit_session)
-disconnected_session := disconnected_$(version)_$(timestamp)
-disconnected_results := $(RESULTSDIR)/$(disconnected_session)
-singlebigip_session := singlebigip_$(version)_$(timestamp)
-singlebigip_results := $(RESULTSDIR)/$(singlebigip_session)
-
-
-testenv_config := bigip.testenv.yaml
-
-VENV := buildbot
-VENV_ACTIVATE := $(MAKEFILE_DIR)/$(VENV)/bin/activate
-
-# Before we run any tests we need a virtualenv and a few packages
-pre-build:
-	sudo -E apt-get update; \
-	sudo -E apt-get install -y libssl-dev; \
-	sudo -E apt-get install -y libffi-dev; \
-	sudo -E -H pip install --upgrade pip; \
-	sudo -E -H pip install tox; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/tools/testenv.git; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/velcro/systest-common.git; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/tools/pytest-meta.git; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/tools/pytest-autolog.git; \
-	sudo -E -H pip install git+ssh://git@bldr-git.int.lineratesystems.com/tools/pytest-symbols.git;
-
-unit: pre-build
-	cd $(MAKEFILE_DIR)/../; \
+TAGINFO := $(shell git describe --long)
+
+# git branch
+BRANCH := $(shell git rev-parse --abbrev-ref HEAD)
+
+# /ASB/PATH/systest/test_results: This is where the buildbot system will look
+# for test reuslts.
+RESULTS_DIR := $(MAKEFILE_DIR)/test_results
+
+
+#### INITIALIZATION SECTION
+#    Since all tests will run on the same buildbot worker, and we do not (yet)
+#    require different packages for different suites, all packages are
+#    installed.  Each deploy maps to its own GUMBALLS project.  This section
+#    has one because unittests are run in this section.  
+#    Tests use tox for python variants.
+functest: 
+	-@echo executing $@
+	sudo -E -H $(SCRIPT_DIR)/install_test_infra.sh &&
+	export GUMBALLS_PROJECT=$(PROJNAME)_$(BRANCH)-unit &&
+	export RESULTS_DIR=$(RESULTS_DIR)/$${GUMBALLS_PROJECT} &&
 	tox -e unit --sitepackages -- \
 		--exclude incomplete no_regression \
-		--autolog-outputdir $(RESULTSDIR) \
-		--autolog-session $(unit_session) \
-
-functest:
-	$(MAKE) -C . unit
-	$(MAKE) -j -C . functest_all
-
-functest_all:
-	@echo "automated functional tests..."
-	$(MAKE) -C . disconnected_service
+		--autolog-outputdir $${RESULTS_DIR} \
```

Comment:
RESULTS_DIR is both a makefile and exported env variable?  Suggest picking a different name for one of them so it's obvious when you choose one over the other.